src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets

```diff
@@ -56,24 +57,49 @@ Copyright (c) .NET Foundation. All rights reserved.
     </TfmSpecificPackageFile>
   </ItemDefinitionGroup>
 
+  <Target Name="_GetOutputItemsFromPack"
+          Returns="@(_OutputPackItems)">
+
+    <ConvertToAbsolutePath Paths="$(PackageOutputPath)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="PackageOutputAbsolutePath" />
+    </ConvertToAbsolutePath>
+
+    <ConvertToAbsolutePath Paths="$(BaseIntermediateOutputPath)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="NuspecOutputAbsolutePath" />
+    </ConvertToAbsolutePath>
+
+    <GetPackOutputItemsTask
+        PackageOutputPath="$(PackageOutputAbsolutePath)"
+        NuspecOutputPath="$(NuspecOutputAbsolutePath)"
+        PackageId="$(PackageId)"
+        PackageVersion="$(PackageVersion)"
+        IncludeSymbols="$(IncludeSymbols)"
+        IncludeSource="$(IncludeSource)">
+
+      <Output
+        TaskParameter="OutputPackItems"
+        ItemName="_OutputPackItems" />
+    </GetPackOutputItemsTask>
+  </Target>
+
   <!--
     ============================================================
     Pack
     Post Build Target
     ============================================================
   -->
   <Target Name="_PackAsBuildAfterTarget"
-          AfterTargets="Build" 
+          AfterTargets="Build"
           Condition="'$(GeneratePackageOnBuild)' == 'true' AND '$(IsInnerBuild)' != 'true'"
           DependsOnTargets="Pack">
   </Target>
 
   <Target Name="_CleanPackageFiles"
+          DependsOnTargets="_GetOutputItemsFromPack"
```

Comment:
❗️ There is no need to manually delete these files. You should instead modify the code which *creates* the package to add the output files to two item groups:

* `FileWrites`
* `_GeneratedCodeFiles`

You can see an example of this [here](https://github.com/tunnelvisionlabs/antlr4cs/blob/7f3d302739417bc2d21f01b3a1012a8af0abef6d/runtime/CSharp/Antlr4BuildTasks/Antlr4.net40.targets#L159-L178), but keep in mind that you should not need to write the intermediate files list unless you need the output of `_GetOutputItemsFromPack` for some other purpose. Obviously you also won't need to add the items to the `Compile` group.